### PR TITLE
pkg/util/tlsutil: add tls primitives functions

### PR DESCRIPTION
**Pull request by fanminshi**
add tls primitives that generate the private key, the ca cert, and the signed cert for the private key.

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hasbro17 @rithujohn191

**Comment by fanminshi** [comment]
fixed CI failure.

**Comment by hasbro17** [comment]
I think you meant for the file name to be `primitives.go` and not `primatives.go`. We can also call them `helpers.go` since they are just helper functions, but I'll leave that up to you.

File: pkg/tlsutil/primatives.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"k8s.io/api/core/v1"
+)
+
+const (
+	rsaKeySize   = 2048
+	duration365d = time.Hour * 24 * 365
+)
+
+// NewPrivateKey returns randomly generated RSA private key.
+func NewPrivateKey() (*rsa.PrivateKey, error) {
```
Comment (hasbro17) [inline pkg/tlsutil/primatives.go]: Do all of these helpers need to be exported? I can't see how they will end up being used by GenerateCert() but I assume it's going to be in the same package. 
So we can keep these functions as lowercase and unexported.
They're not meant to be exposed to the user.

Comment (fanminshi) [reply pkg/tlsutil/primatives.go]: yeah, you are right. they are not meant to be exported.

**Comment by fanminshi** [comment]
I kept the `primitives.go` convention. And make the functions private.

**Comment by hasbro17** [comment]
LGTM

**Pull request merged**